### PR TITLE
pytorch: fix cve-2024-31580, cve-2024-31583, cve-2024-27319

### PR DESCRIPTION
**Pull request by ddstreetmicrosoft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
PAtches 'pytorch' package for cve-2024-31580, cve-2024-31583, cve-2024-27319

###### Change Log  

PAtches 'pytorch' package for cve-2024-31580, cve-2024-31580, cve-2024-27319
- Change
- Change
- Change

###### Does this affect the toolchain?  

/NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/cve-2024-31580
- https://nvd.nist.gov/vuln/detail/cve-2024-31583
- https://nvd.nist.gov/vuln/detail/cve-2024-27319

###### Test Methodology

- Pipeline build id: xxxx
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=555643&view=results

### CODE REVIEW

**Review by rikenm1 — APPROVED** [review]
LGTM

**Review by PawelWMS — APPROVED** [review]

File: SPECS/pytorch/pytorch.spec [hunk 1]
```diff
@@ -11,6 +11,10 @@ URL:            https://pytorch.org/
 Source0:        https://github.com/pytorch/pytorch/releases/download/v%{version}/%{name}-v%{version}.tar.gz#/%{name}-%{version}.tar.gz
 # Use the generate_source_tarball.sh script to create a tarball of submodules during version updates.
 Source1:        %{name}-%{version}-submodules.tar.gz
+Patch0:         CVE-2024-31580.patch
```
[the remark this answers is not in the document]
Comment (PawelWMS) [reply SPECS/pytorch/pytorch.spec:14]: But feel free to merge into `main` first - just keep in mind that there's a chance more work may need to be done for a `fasttrack/2.0` -> `main` cherry-pick depending on the differences between the two branches. Happens rarely.

**Review by PawelWMS — APPROVED** [review]

File: SPECS/pytorch/pytorch.spec (hunk 1 above)
[the remark this answers is not in the document]
Comment (ddstreetmicrosoft) [reply SPECS/pytorch/pytorch.spec:14]: i was told fasttrack is closed this week which i why i switched it from fasttrack over to main

[the remark this answers is not in the document]
Comment (PawelWMS) [reply SPECS/pytorch/pytorch.spec:14]: I think it's only closed until the 2.0 release is published. It hope it won't be a whole week but Jon would know better. Once that's done the critical and high CVEs, which didn't make it with the release, need to be published through fast-track, though.

**Pull request merged**